src/python_testing/matter_testing_support.py

```diff
@@ -1492,19 +1492,35 @@ def parse_matter_test_args(argv: Optional[List[str]] = None) -> MatterTestConfig
 
     return convert_args_to_matter_config(parser.parse_known_args(argv)[0])
 
+def async_to_sync(awaitable):
+    "Call sync, the async function."
+    # Usage of new implementation of ChipDeviceController cause the necessary to not use asyncio.run,
+    # because the initialization of `asyncio.Lock` use the `asyncio.get_event_loop()` on init.
+    # Usage of `asyncio.run()` and `asyncio.get_event_loop()` are exclusive.
```

Comment:
It's not 100% clear why this change is needed.

Reply:
The usage of `asyncio.run()` and  `asyncio.get_event_loop()` are mutually exclusive. 
In python 3.9.18 creation of asyncio.Lock call  `asyncio.get_event_loop()`:
https://github.com/python/cpython/blob/3.9/Lib/asyncio/locks.py#L81

Example of reproduction:
```
import asyncio

async def hello():
    print("hello")

lock_1 = asyncio.Lock()
asyncio.run(hello())
lock_2 = asyncio.Lock()
```

Reply:
I think that I will fix that issue on my side, while using the scripts.
Since it should be an issue in the Python 3.9 and older version.